### PR TITLE
Quickstart branch

### PR DESCRIPTION
**Pull request by sergiolaverde0**
To merge this in, please create a new branch off your current main without any changes, and then let me know you did so I change the target branch.

I have tested it from the side of a new user and it works great, they still have to use `chmod 777` because I wasn't able to figure out a better workaround for the permission issues across the backend but at least there is no need for `sudo`.

Old users will need `sudo` in their already existing files, but at least this way they will be able to move the files from the old locations to the new ones using a graphical file explorer instead of the commandline if they want. I was also able to confirm that the old layout was unable to import epub files before but it works just fine after.

I will be opening another PR for the changes to the `README.md` later, but is basically `chmod 777 -R ./` inside the cloned branch, and the same but with `sudo` in the already existing install, always before running the new containers for the first time and while the old containers are down if they exist.

### CODE REVIEW

**Comment by simjanos-dev** [comment]
Thank you! 

I'll merge this as you requested. There is still two problems: the flags and the epub cover files that will be lost for already existing users.

I've just tested my last code I want to add to v0.5. I'll commmit it in, move the flags and write the migration file to reset cover images to default. I think it should take around 30~ minutes. After that I'll test your 2 PR-s. I will merge #71 into dev after testing it, since there is no need to do it as a hotfix now.

Please only modify the readme in the current dev branch, because I've also changed it. I've tested all my changes multiple times, so if the docker part seems good, we can merge it in today(It's 2PM at me).

**Comment by sergiolaverde0** [comment]
Gimme 30 minutes to see if I can rescue the epub cover files with `docker cp`. And what's wrong with the flags?

**Comment by simjanos-dev** [comment]
Flags were in the `storage/app/images/flags` directory. With this PR the whole `/storage` directory will be overwritten. I think we should store the flags in the main repo, and copy it into the images, they shouldn't be on a a mounted directory.

**Comment by sergiolaverde0** [comment]
I was able to recover the book covers. Also, flags *seem* to work normally for me even with this setup, they just popped into the mounted volume. I confess not being sure about the reason.

**Comment by simjanos-dev** [comment]
That is great news, thank you!

I'll move the flags anyway. I think the public directory a better place for them, as they are constant non changing images.

**Comment by sergiolaverde0** [comment]
For you to test the cover recovery without waiting for me to curate a commit, the command is just `docker cp linguacafe-webserver:/var/www/html/storage/app/images/book_images ./` and then you will have the folder in the same location you ran the command.

**Comment by simjanos-dev** [comment]
Thank you! I'll wait for you with that. I'll test #71, it will take a while.

**Comment by simjanos-dev** [comment]
We should put the instructions for this cover image saving to a very visible place, because the program won't function properly if someone upgrades and it doesn't find the images.

**Comment by simjanos-dev** [comment]
Okay. My final changes are in the `dev` branch for v0.5. The only thing left is docker, readme and uploading new dictionaries.

I've also noticed that I need root permission to move my database folder around. I'm using a dev environment, but the mysql part is same.

**Comment by sergiolaverde0** [comment]
> I've also noticed that I need root permission to move my database folder around.

Yes, this is what made `sudo` unavoidable for existing installs.

I have made a last commit renaming a folder to ensure no folders change names to make migration as smooth as possible, and the PR with the migration instructions is up.

File: mysql/.gitkeep [hunk 1]
```diff
@@ -0,0 +1,4 @@
+  ／l、?             
+（ﾟ､ ｡ ７         
+  l  ~ヽ       
+  じしf_,)ノ
```
Comment (simjanos-dev) [inline mysql/.gitkeep:4]: I think we should create a storage and a database empty directory with this .gitkeep file.

File: docker-compose.yml [hunk 2]
```diff
@@ -31,7 +29,8 @@ services:
         ports:
             - "3308:3306"
         volumes:
-            - /your/linguacafe/database/folder:/var/lib/mysql
+            - ./database:/var/lib/mysql
+        user: 1000:1000
```
Comment (simjanos-dev) [inline docker-compose.yml:33]: Will this `user: 1000:1000` affect already existing database folders?

**Comment by simjanos-dev** [comment]
How can we test this? I don't think mergin into the main is a good idea, because building the image for github takes 30 minutes, I need 30~ minutes to download the image, because my laptop always freezes for quite a few minutes. 

Can you duplicate the image building to the `dev` branch with `docker-compose-dev-deploy.yml` file?

I'll go and create a deploy branch.

**Comment by sergiolaverde0** [comment]
I have now changed the target branch. This will never touch `main`, the branch will exist sorely as a way to keep users from creating directories manually. I have also noticed oversights both here and on the other PR; I will fix them immediately.

File: docker-compose.yml (hunk 2 above)
Comment (sergiolaverde0) [reply docker-compose.yml:33]: It was harmless when tested, but it is still an oversight that will be fixed.

**Comment by simjanos-dev** [comment]
Oh, I understand that. 

I just wanted to create a process, that works the same as deploying to main, just instead deploying the dev with a `:dev` image tag instead of `:main`.  This could be used as a final test. I don't know if it's necessary or not. If docker will work, I won't need it because there wouldn't be any changes to it.

**Comment by sergiolaverde0** [comment]
I get you now. You basically need an action, preferably one that is triggered manually, and that builds a test image off `dev`. Make it target amd64 only and skip arm64 so that it builds faster.

Give me some time and I will do it.

**Comment by simjanos-dev** [comment]
Yes, exactly. Thank you so much! Im pulling the race condition fix now. Think it will be 20~ minutes.

**Pull request merged**